app/models/token.rb

```diff
@@ -23,4 +23,20 @@ def self.create_from_omniauth(credentials, user)
   def expired?
     expiration < Time.zone.now
   end
+
+  # Checks is a token is still valid with ORCiD
+  # A token can become invalid if the user revokes it within ORCiD
+  def self.valid_in_orcid?(token, orcid)
+    url = "https://api.sandbox.orcid.org/v3.0/#{orcid}/record"
```

Comment:
This URL should be configurable, I documented this on a new issue: https://github.com/pulibrary/orcid_princeton/issues/240